daml-lf/archive/BUILD.bazel

```diff
@@ -171,6 +171,7 @@ da_scala_test_suite(
 daml_compile_with_dalf(
     name = "DarReaderTest",
     srcs = ["src/test/daml/DarReaderTest.daml"],
+    target = lf_version_configuration.get("latest"),
```

Comment:
We uses this target in 4 test: 
1. `DarReaderTest` 
2. `UniversalArchiveReaderSpec`
3. `ZipBombDetectionSpec`
4. `StablePackahes`

the 3 first one test `DarReader` and `UniversalArchiveReader` which are helper objects to read dalf/dar. They are not so concern about the actual content of the dar. Checking the decoder handle the versions properly is done by other unit tests. 

For 4, i need a dar compiler with the more up-to-date version of LF, i.e. stable.

Reply:
I see it a bit differently, 1. and 2. should use `DarReaderTest` as it is (without the explicit `target`, equivalent to `target = lf_version_configuration.get("default")`) since, like you said, they are not concerned about the actual contents. It looks like you've split 3. into individual cases in 1. and 2., so that's covered as well. 

Finally, for 4. I think we would actually want to test multiple LF versions, at the very least 1.14, 1.15 and 1.dev. I don't think the default/stable/latest labels help us much, since stable packages define their first LF version using concrete numbers. I'm thinking of something like

```starlark
[
    daml_compile_with_dalf(
        name = "StablePackageTest-%s" % ver,
        srcs = ["src/test/daml/DarReaderTest.daml"], # i think it's fine to reuse the source file
        target = ver,
    )
    for ver in COMPILER_LF_VERSIONS
]
```

then in the test we check that only the stable packages such that `stablePackage.languageVersion >= dar.main.languageVersion` are present